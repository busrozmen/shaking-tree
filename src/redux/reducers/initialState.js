 const initialState = {
  isShaked: false,
  progress: false
}
export default initialState