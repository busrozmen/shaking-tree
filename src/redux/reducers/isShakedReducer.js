
import * as actionTypes from '../actions/actionTypes'
import initialState from './initialState'

export default function isShakedReducer ( state=initialState.isShaked, action ) {
  switch (action.type) {
    case actionTypes.IS_SHAKED:
      return action.payload
    default:
      return state;
  }
}