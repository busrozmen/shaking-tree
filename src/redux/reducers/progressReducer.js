

import * as actionTypes from '../actions/actionTypes'
import initialState from './initialState'

export default function progressReducer (state=initialState.progress, action ) {
  switch (action.type) {
    case actionTypes.GET_PROGRESS:
      return action.payload
    default:
      return state;
  }
}