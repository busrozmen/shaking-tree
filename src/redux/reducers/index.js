import {combineReducers} from 'redux'
import isShakedReducer from './isShakedReducer'
import progressReducer from './progressReducer'

const rootReducer = combineReducers({
  isShakedReducer,
  progressReducer
})

export default rootReducer;