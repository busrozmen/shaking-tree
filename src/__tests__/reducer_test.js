import progressReducer from '../redux/reducers/progressReducer'
import isShakedReducer from '../redux/reducers/isShakedReducer'
import {initialState} from '../redux/reducers/initialState'

describe('progress reducer', () => {
  it('should return the initial state', () => {
    expect(progressReducer(undefined, {})).toMatchSnapshot()
  })
  it('shold handle GET_PROGRESS', () => {
    expect(progressReducer(initialState, 
      {
        type:'GET_PROGRESS'
      })
    ).toMatchSnapshot()
  })
})
describe('shaked reducer', () => {
  it('should return the initial state', () => {
    expect(isShakedReducer(undefined, {})).toMatchSnapshot()
  })
  it('shold handle IS_SHAKED', () => {
    expect(isShakedReducer(initialState, 
      {
        type:'IS_SHAKED'
      })
    ).toMatchSnapshot()
  })
});