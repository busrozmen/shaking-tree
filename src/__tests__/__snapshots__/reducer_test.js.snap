// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress reducer shold handle GET_PROGRESS 1`] = `undefined`;

exports[`progress reducer should return the initial state 1`] = `false`;

exports[`shaked reducer shold handle IS_SHAKED 1`] = `undefined`;

exports[`shaked reducer should return the initial state 1`] = `false`;
